ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -96,20 +97,19 @@ func getClusterType(logger logr.Logger) bool {
 
 // NewReconciler returns a new reconcile.Reconciler
 func NewReconciler(mgr manager.Manager, options RayClusterReconcilerOptions) *RayClusterReconciler {
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Pod{}, podUIDIndexField, func(rawObj client.Object) []string {
+	ctx := context.Background()
+	if err := mgr.GetFieldIndexer().IndexField(ctx, &corev1.Pod{}, podUIDIndexField, func(rawObj client.Object) []string {
```

Comment:
Using the root context would be more correct I think.

Reply:
> root context

Do you mean creating a context in [main.go](https://github.com/ray-project/kuberay/blob/master/ray-operator/main.go) and then passing the context to `NewReconciler`, `NewRayServiceReconciler`, and `NewRayJobReconciler`?

Reply:
Thanks!

Reply:
Not sure whether https://github.com/ray-project/kuberay/pull/1945/commits/145cb7bef3ec5def123fd51cf296848a8f0d9d03 is what you are referring to or not.

Reply:
Yes, exactly that. Like that termination signals are correctly propagated to controllers.